client-state-machine.go

```diff
@@ -59,12 +59,13 @@ type ClientStateStart struct {
 	helloRetryRequest *HandshakeMessage
 }
 
-func (state ClientStateStart) Next(hm *HandshakeMessage) (HandshakeState, []HandshakeAction, Alert) {
-	if hm != nil {
-		logf(logTypeHandshake, "[ClientStateStart] Unexpected non-nil message")
-		return nil, nil, AlertUnexpectedMessage
-	}
+var _ HandshakeState = &ClientStateStart{}
 
+func (ClientStateStart) State() State {
```

Comment:
Aesthetically, I would prefer to have a name for the target object, even if it's not used.